### PR TITLE
[FEATURE] Add HtmlPruner::removeRedundantClassesAfterCssInlined

### LINKED ISSUE #380: remove the class="xxx" from html after converting to inline-css
the classes are perfect converted, but the class is still present in the html tag:

`<span class="f3" style="color: #AA5500;">`

this blows up the output size.

A switch like

`private $shouldClassesBeRemoved = true;`

would be fine.

### PR DESCRIPTION
**Pull request by JakeQZ**
Also added section for `HtmlPruner` to the README.

Closes #380.

### CODE REVIEW

File: tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php [hunk 1]
```diff
@@ -359,6 +360,182 @@ public function removeRedundantClassesMinifiesClassAttributes($html, array $clas
...
+     * @param string[] $classesToBeKept
+     *
+     * @dataProvider provideClassesInUninlinableRules
+     */
+    public function removeRedundantClassesAfterCssInlinedNotRemovesClassesInUninlinableRules(
+        $html,
+        $css,
+        array $classesToBeKept = []
+    ) {
+        \extract($this->buildSubjectAndCssInlinerWithCssInlined('<html>' . $html . '</html>', $css));
```
Comment (oliverklee) [inline tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php]: Please let's not use `extract` and `compact` as static code analyzers have a really hard time with these functions. Instead, please let's use `list` and explicit arrays.

File: tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php [hunk 2]
```diff
@@ -359,6 +360,182 @@ public function removeRedundantClassesMinifiesClassAttributes($html, array $clas
...
+
+        $subject->removeRedundantClassesAfterCssInlined($cssInliner);
+
+        $result = $subject->render();
+        foreach ($classesToBeKept as $class) {
+            self::assertContains($class, $result);
```
Comment (oliverklee) [inline tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php]: I propose we add a `private static assertContainsAll` method for this (and `assertContainsNone`).

File: tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php (hunk 1 above)
Comment (JakeQZ) [reply tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php]: I was wondering what you'd think about that and aware that using `list` was an alternative solution to the 'multiple return values problem'.  Will change accordingly...

File: tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php (hunk 2 above)
Comment (JakeQZ) [reply tests/Unit/Emogrifier/HtmlProcessor/HtmlPrunerTest.php]: There might, from memory, be tests in #708 that could also use this.  Will see...

**Comment by JakeQZ** [comment]
Have made suggested changes and repushed...

**Comment by JakeQZ** [comment]
> Looks good! Could you please rebase to resolve the conflict?

Have done.  (Now forewarned and forearmed, `git rebase master` worked seamlessly, with forced push.)

**Pull request merged**